src/data.rs

```diff
@@ -24,6 +24,14 @@ impl VariantData {
             VariantData::Unit => &[],
         }
     }
+
+    pub fn fields_mut(&mut self) -> &mut [Field] {
+        match *self {
+            VariantData::Struct(ref mut fields) |
+            VariantData::Tuple(ref mut fields) => fields,
+            VariantData::Unit => &mut [],
```

Comment:
I was shocked that this worked. I get that it's completely valid from a memory safety point of view since there's no operation that could actually mutate an empty slice, but I didn't think the compiler would special case empty slices vs non-empty here.